In [161]:
import pandas as pd
import numpy as np

# 2007-2008 Data

In [162]:
# 2007-2008
path = '/Users/carolinesklaver/Desktop/Capstone/NHANES/data/csv_data/2007-2008'

import os
os.chdir(path)

## PHQ-9 Survey - part of Questionnaire

In [163]:
DPQ_2007_2008 = pd.read_csv('./Questionnaire/DPQ_E.csv')

In [164]:
DPQ_2007_2008.head()

SEQN        DPQ010        DPQ020        DPQ030        DPQ040  \
0  41475.0  1.000000e+00  5.397605e-79  1.000000e+00  1.000000e+00   
1  41477.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
2  41479.0  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3  41481.0  5.397605e-79  5.397605e-79  1.000000e+00  1.000000e+00   
4  41482.0  9.000000e+00  1.000000e+00  1.000000e+00  5.397605e-79   

         DPQ050        DPQ060        DPQ070        DPQ080        DPQ090  \
0  1.000000e+00  1.000000e+00  5.397605e-79  5.397605e-79  5.397605e-79   
1  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
2  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
3  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   
4  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79  5.397605e-79   

         DPQ100  
0  5.397605e-79  
1           NaN  
2           NaN  
3  5.397605e-79  
4  5.397605e-79

In [165]:
# drop final follow-up question
DPQ_2007_2008.drop(['DPQ100'], axis=1, inplace=True)

In [166]:
# values less than 1 are zero
def set_zero(df):
    df[df<1] = 0
    
set_zero(DPQ_2007_2008)

In [167]:
# need to drop rows where not all col are complete
DPQ_2007_2008.dropna(axis=0, how='any', thresh=None, subset=None, inplace=True)

In [168]:
# take a sum across rows and save as new col
DPQ_2007_2008['sum'] = DPQ_2007_2008.iloc[:,1:].sum(axis=1)

In [169]:
# create new target label 'depressed' if sum is greater than 10 --> 1, otherwise 0
DPQ_2007_2008.loc[DPQ_2007_2008['sum'] > 10, 'depressed'] = 1
DPQ_2007_2008.loc[DPQ_2007_2008['sum'] <= 10, 'depressed'] = 0

In [170]:
DPQ_2007_2008.head()

SEQN  DPQ010  DPQ020  DPQ030  DPQ040  DPQ050  DPQ060  DPQ070  DPQ080  \
0  41475.0     1.0     0.0     1.0     1.0     1.0     1.0     0.0     0.0   
1  41477.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2  41479.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3  41481.0     0.0     0.0     1.0     1.0     0.0     0.0     0.0     0.0   
4  41482.0     9.0     1.0     1.0     0.0     0.0     0.0     0.0     0.0   

   DPQ090   sum  depressed  
0     0.0   5.0        0.0  
1     0.0   0.0        0.0  
2     0.0   0.0        0.0  
3     0.0   2.0        0.0  
4     0.0  11.0        1.0

In [171]:
DPQ_2007_2008.shape

(5447, 12)

In [172]:
DPQ_E = DPQ_2007_2008[['SEQN', 'depressed']]

In [173]:
DPQ_E.head()

SEQN  depressed
0  41475.0        0.0
1  41477.0        0.0
2  41479.0        0.0
3  41481.0        0.0
4  41482.0        1.0

In [174]:
len(DPQ_2007_2008[DPQ_2007_2008['depressed']==1])  # about 8% depressed = true

455

## Demographic Data

### Demographic Variables and Sample Weights (DEMO_E)

In [175]:
# start with demographic data
DEMO_2007_2008 = pd.read_csv('./Demographics/DEMO_E.csv')

In [176]:
DEMO_2007_2008.head(10)

SEQN  SDDSRVYR  RIDSTATR  RIDEXMON  RIAGENDR      RIDAGEYR  RIDAGEMN  \
0  41475.0       5.0       2.0       2.0       2.0  6.200000e+01     751.0   
1  41476.0       5.0       2.0       1.0       2.0  6.000000e+00      81.0   
2  41477.0       5.0       2.0       2.0       1.0  7.100000e+01     859.0   
3  41478.0       5.0       2.0       2.0       2.0  1.000000e+00      17.0   
4  41479.0       5.0       2.0       1.0       1.0  5.200000e+01     629.0   
5  41480.0       5.0       2.0       1.0       1.0  6.000000e+00      81.0   
6  41481.0       5.0       2.0       1.0       1.0  2.100000e+01     254.0   
7  41482.0       5.0       2.0       2.0       1.0  6.400000e+01     778.0   
8  41483.0       5.0       2.0       1.0       1.0  6.600000e+01     803.0   
9  41484.0       5.0       2.0       1.0       1.0  5.397605e-79       9.0   

   RIDAGEEX  RIDRETH1  DMQMILIT  ...  FIAPROXY  FIAINTRP  MIALANG  MIAPROXY  \
0     752.0       5.0       2.0  ...       2.0       2.0      1.0       2.0   
1      82.0       5.0       NaN  ...       2.0       2.0      NaN       NaN   
2     860.0       3.0       1.0  ...       2.0       2.0      1.0       2.0   
3      17.0       3.0       NaN  ...       2.0       2.0      NaN       NaN   
4     630.0       1.0       2.0  ...       2.0       2.0      2.0       2.0   
5      81.0       1.0       NaN  ...       2.0       2.0      NaN       NaN   
6     254.0       4.0       2.0  ...       2.0       2.0      1.0       2.0   
7     779.0       1.0       2.0  ...       2.0       2.0      2.0       2.0   
8     804.0       4.0       1.0  ...       2.0       2.0      1.0       2.0   
9      10.0       3.0       NaN  ...       2.0       2.0      NaN       NaN   

   MIAINTRP  AIALANG      WTINT2YR      WTMEC2YR  SDMVPSU  SDMVSTRA  
0       2.0      1.0  59356.356426  60045.772497      1.0      60.0  
1       NaN      NaN  35057.218405  35353.210440      1.0      70.0  
2       2.0      1.0   9935.266183  10074.150074      1.0      67.0  
3       NaN      NaN  12846.712058  14560.472652      2.0      59.0  
4       2.0      2.0   8727.797555   9234.055759      1.0      70.0  
5       NaN      NaN   7379.745086   7297.067503      2.0      69.0  
6       2.0      1.0  24342.505253  24655.376656      2.0      68.0  
7       2.0      2.0   9811.075078  11602.178638      2.0      65.0  
8       2.0      1.0   8058.685296   7920.812275      2.0      66.0  
9       NaN      NaN   8942.951928   9259.270099      2.0      59.0  

[10 rows x 43 columns]

Demographics variables:
- DMDBORN4: Due to the concerns of disclosure risk, starting in 2011, country of birth was recoded into two categories: 1) Born in 50 U.S. states or Washington, DC; and 2) Born in other countries, including U.S. territories.

- DMDEDUC2: This variable is the highest grade or level of education completed by adults 20 years and older. The response categories are: less than 9th grade education, 9-11th grade education (includes 12th grade and no diploma), High school graduate/GED, some college or associates (AA) degree, and college graduate or higher.

- DMDHHSIZ: This variable is the number of people in the participant’s household. The values for this variable range from 1 to 7. Due to disclosure concerns, households that are comprised of 7 or more people are included in the category that is labeled ‘7 or more’.

- RIDAGEYR: Age in years, at the time of the screening interview, is reported for survey participants between the ages of 1 and 79 years of age. All responses of participants aged 80 years and older are coded as ‘80.’

- RIAGENDR: Gender: male 1, female 2

- INDHHIN2: Total household income (reported as a range value in dollars), range 1-15, 77 refused, 99 dont know

- DMQMIL1Z: Served active duty in US Armed Forces, 1 yes, 0 no, 7 refuse, 9 dont know

- DMDMARTL: Marital status - 1 Married, 2 widowed, 3 divorced, 4 separated, 5 never married, 6 living with partner, 77 refused	


In [177]:
# filter only relevant cols
DEMO_E = DEMO_2007_2008.filter(['SEQN','RIDRETH1', 'DMDEDUC2', 'DMDHHSIZ', 'RIDAGEYR', 'RIAGENDR','INDHHIN2', 
                        'DMQMIL1Z', 'DMDMARTL'], axis=1)

In [178]:
# rename col to be more useful
demo_cols = {'SEQN': 'SEQN', 'RIDRETH1':'race_ethnicity', 'DMDEDUC2': 'edu_level', 'DMDHHSIZ': '#_ppl_household', 
               'RIDAGEYR': 'age', 'RIAGENDR': 'gender','INDHHIN2': 'annual_HI', 
                        'DMQMIL1Z': 'served_army', 'DMDMARTL':'marital_status'}

DEMO_E.columns = DEMO_E.columns.map(demo_cols)

In [179]:
DEMO_E.head()

SEQN  race_ethnicity  edu_level  #_ppl_household   age  gender  \
0  41475.0             5.0        3.0              2.0  62.0     2.0   
1  41476.0             5.0        NaN              6.0   6.0     2.0   
2  41477.0             3.0        3.0              2.0  71.0     1.0   
3  41478.0             3.0        NaN              4.0   1.0     2.0   
4  41479.0             1.0        1.0              5.0  52.0     1.0   

   annual_HI  marital_status  
0        6.0             1.0  
1       15.0             NaN  
2        5.0             1.0  
3        3.0             NaN  
4        8.0             1.0

## Dietary Data

### Dietary Interview - Total Nutrient Intakes, First Day (DR1TOT_E)

In [180]:
DIET_2007_2008 = pd.read_csv('./Dietary/DR1TOT_E.csv')

Dietary variables:
- DR1TCAFF : caffeine - Caffeine (mg) range 0 to 2448
- DRQSDIET - On special diet? - 1,2,9, missing
- DR1TPROT - Protein (gm) - 0 to 869.49, missing
- DR1TCARB - Carbohydrate (gm) - 8.67 to 1423.87, missing
- DR1TSUGR - Total sugars (gm) - 0.13 to 1115.5, missing
- DR1TFIBE - Dietary fiber (gm) - 0 to 136.6, missing
- DR1TTFAT - Total fat (gm) - 0 to 548.38, missing
- DR1TCHOL - Cholesterol (mg) - 0 to 3515, missing
- DR1TVB6 - Vitamin B6 (mg)
- DR1TFOLA - Total folate (mcg) - 0 to 3172, missing
- DR1TVB12 - Vitamin B12 (mcg) - 0 to 110, missing
- DR1TVD - Vitamin D (D2 + D3) (mcg) - 0 to 86.3, missing
- DR1TMAGN - Magnesium (mg) - 0 2725, missing
- DR1TIRON - Iron (mg) - 0 to 130, missing
- DR1TZINC - Zinc (mg) - 0 to 116, missing
- DR1TSELE - Selenium (mcg) - 0 to 986.5, missing
- DR1_320Z - Total plain water drank yesterday (gm) - 0 13440, missing

In [181]:
# filter only relevant cols
DIET_E = DIET_2007_2008.filter(['SEQN', 'DR1TCAFF', 'DRQSDIET', 'DR1TPROT','DR1TCARB','DR1TSUGR','DR1TFIBE',
                               'DR1TTFAT','DR1TCHOL','DR1TVB6','DR1TFOLA','DR1TVB12','DR1TVD',
                               'DR1TMAGN','DR1TIRON','DR1TZINC','DR1TSELE','DR1_320Z'], axis=1)

diet_cols = {'SEQN': 'SEQN', 'DR1TCAFF':'caffeine','DRQSDIET': 'special_diet', 'DR1TPROT':'protein',
             'DR1TCARB':'carb','DR1TSUGR':'sugar','DR1TFIBE':'fiber',
            'DR1TTFAT':'fat','DR1TCHOL':'diet_chol','DR1TVB6':'VitB6','DR1TFOLA':'folate',
             'DR1TVB12':'VitB12','DR1TVD':'VitD', 'DR1TMAGN':'magnesium',
             'DR1TIRON':'iron','DR1TZINC':'zinc','DR1TSELE':'selenium','DR1_320Z':'water'}

DIET_E.columns = DIET_E.columns.map(diet_cols)

In [182]:
DIET_E.head()

SEQN  caffeine  special_diet  protein    carb   sugar  fiber     fat  \
0  41475.0     549.0           1.0    62.78  192.19   83.35   10.0   65.64   
1  41476.0      21.0           2.0    69.38  276.93  167.71   14.4   82.47   
2  41477.0     188.0           2.0    40.90   94.78   18.25    5.6   42.56   
3  41478.0      26.0           2.0    38.52  161.69   92.13   10.7   40.25   
4  41479.0       3.0           2.0   138.85  319.65  123.06   32.7  125.96   

   diet_chol  VitB6  folate  VitB12  VitD  magnesium   iron   zinc  selenium  \
0      587.0  1.076   366.0    3.59   3.0      244.0  12.59   9.50     113.6   
1      362.0  1.624   463.0    5.85   8.0      265.0  10.10  10.52      87.0   
2      401.0  0.606   251.0    1.56   1.4       99.0   9.09   5.77      68.2   
3       89.0  1.204   210.0    3.25   5.6      175.0  10.92   6.38      51.6   
4      461.0  2.697   510.0    6.73   2.9      530.0  22.88  21.63     163.8   

          water  
0  1.007250e+03  
1  2.370000e+02  
2  5.397605e-79  
3  1.481300e+02  
4  7.110000e+02

## Questionnaire Data

### Diabetes

variables
- DIQ010 : doc_diabetes - {Have you/Has SP}} ever been told by a doctor or health professional that {you have/{he/she/SP} has} diabetes or sugar diabetes? 1 yes, 2 no, 3 boarderline
- DIQ160 : doc_prediabetes - {Have you/Has SP} ever been told by a doctor or other health professional that {you have/SP has} any of the following: prediabetes, impaired fasting glucose, impaired glucose tolerance, borderline diabetes or that {your/her/his} blood sugar is higher than normal but not high enough to be called diabetes or sugar diabetes? 1 yes, 2 no

In [183]:
DIQ_2007_2008 = pd.read_csv('./Questionnaire/DIQ_E.csv')

In [184]:
DIQ_2007_2008.columns

Index(['SEQN', 'DIQ010', 'DID040', 'DIQ220', 'DIQ160', 'DIQ170', 'DIQ180',
       'DIQ190A', 'DIQ190B', 'DIQ190C', 'DIQ200A', 'DIQ200B', 'DIQ200C',
       'DIQ050', 'DID060', 'DIQ060U', 'DID070', 'DIQ230', 'DIQ240', 'DID250',
       'DID260', 'DIQ260U', 'DID270', 'DIQ280', 'DIQ290', 'DIQ300S', 'DIQ300D',
       'DID310S', 'DID310D', 'DID320', 'DID330', 'DID340', 'DID350', 'DIQ350U',
       'DIQ360', 'DIQ080'],
      dtype='object')

In [185]:
DIQ_E = DIQ_2007_2008.filter(['SEQN','DIQ010', 'DIQ160'], axis=1)

In [186]:
# rename col to be more useful
DIQ_cols = {'SEQN':'SEQN', 'DIQ010':'doc_diabetes', 'DIQ160': 'doc_prediabetes'}

DIQ_E.columns = DIQ_E.columns.map(DIQ_cols)

In [187]:
DIQ_E.head()

SEQN  doc_diabetes  doc_prediabetes
0  41475.0           2.0              2.0
1  41476.0           2.0              NaN
2  41477.0           1.0              NaN
3  41478.0           2.0              NaN
4  41479.0           2.0              2.0

### Diet Questionnaire

Variables:
- CBQ505 : fast_Eood_12m - In the past 12 months, did you buy food from fast food or pizza places?
- DBD895 : #meals_not_Eome - During the past 7 days, how many meals {did you/did SP} get that were prepared away from home in places such as restaurants, fast food places, food stands, grocery stores, or from vending machines?
    - 0 None, 1 to 21 range, 5555 more than 21 meals per week, 7777 refused, 9999 dont know
- DBD900 : #meals_East_Eood - How many of those meals {did you/did SP} get from a fast-food or pizza place?
- DBQ700 : how_Eealthy_diet - In general, how healthy is {your/his/her} overall diet? 
    - Excellent to Poor: 1-5

In [188]:
DBQ_2007_2008 = pd.read_csv('./Questionnaire/DBQ_E.csv')

In [189]:
DBQ_2007_2008.columns

Index(['SEQN', 'DBQ010', 'DBD020', 'DBD030', 'DBD040', 'DBD050', 'DBD060',
       'DBD072A', 'DBD072B', 'DBD072C', 'DBD072D', 'DBD072U', 'DBD080',
       'DBQ700', 'DBQ197', 'DBD222A', 'DBD222B', 'DBD222C', 'DBD222D',
       'DBD222U', 'DBQ229', 'DBQ235A', 'DBQ235B', 'DBQ235C', 'DBQ301',
       'DBQ330', 'DBQ360', 'DBQ370', 'DBD381', 'DBQ390', 'DBQ400', 'DBD411',
       'DBQ421', 'DBQ424', 'DBD895', 'DBD900', 'DBD905', 'DBD910', 'DBQ915',
       'DBQ920', 'DBQ925A', 'DBQ925B', 'DBQ925C', 'DBQ925D', 'DBQ925E',
       'DBQ925F', 'DBQ925G', 'DBQ925H', 'DBQ925I', 'DBQ925J', 'DBQ930',
       'DBQ935', 'DBQ940', 'DBQ945'],
      dtype='object')

In [190]:
DBQ_E = DBQ_2007_2008.filter(['SEQN','DBD900', 'DBQ700'], axis=1)

In [191]:
# rename col to be more useful
DBQ_cols = {'SEQN':'SEQN',
                  'DBD900':'#meals_fast_food', 'DBQ700': 'how_healthy_diet'}

DBQ_E.columns = DBQ_E.columns.map(DBQ_cols)

In [192]:
DBQ_E.head()

SEQN  #meals_fast_food  how_healthy_diet
0  41475.0               NaN               3.0
1  41476.0      5.397605e-79               NaN
2  41477.0      1.000000e+00               1.0
3  41478.0      1.000000e+00               NaN
4  41479.0      5.397605e-79               4.0

### Sight & Hearing

variables:
- DLQ010 : difficult_Eearing - 1 yes, 2 no
- DLQ020 : difficult_seeing - 1 yes, 2 no
- DLQ050 - Have serious difficulty walking ? 
    - 1=yes, 2=no, 7/9
- DLQ080 - Have difficulty doing errands alone ? 
    - 1=yes, 2=no, 7/9


In [193]:
#DLQ_2007_2008 = pd.read_csv('./Questionnaire/DLQ_E.csv')

In [194]:
# DLQ_2007_2008.columns

In [195]:
# DLQ_E = DLQ_2007_2008.filter(['SEQN','DLQ010','DLQ020','DLQ050','DLQ080'], axis=1)

In [196]:
# rename col to be more useful
# DLQ_cols = {'SEQN':'SEQN', 'DLQ010': 'difficult_Eearing', 'DLQ020': 'difficult_seeing',
#           'DLQ050':'difficult_walking','DLQ080':'difficult_errands'}

# DLQ_E.columns = DLQ_E.columns.map(DLQ_cols)

In [197]:
# DLQ_E.head()

### Drugs

variables: 
- DUQ211 : used_marijuana - Have you ever smoked marijuana or hashish at least once a month for more than one year?
- DUQ230 : # days used marijuana or hashish/month
- DUQ240 : used_CMH - Have you ever used cocaine, crack cocaine, heroin, or methamphetamine?
- DUQ370 : injected_drugs - Have you ever, even once, used a needle to inject a drug not prescribed by a doctor?

    - 1 yes, 2 no, 7 refused, 9 don't know

In [202]:
DUQ_2007_2008 = pd.read_csv('./Questionnaire/DUQ_E.csv')

In [203]:
DUQ_E = DUQ_2007_2008.filter(['SEQN', 'DUQ240'], axis=1)

In [204]:
DUQ_cols = {'SEQN':'SEQN', 'DUQ240':'used_CMH'}

DUQ_E.columns = DUQ_E.columns.map(DUQ_cols)

In [205]:
DUQ_E.head()

SEQN  used_CMH
0  41475.0       2.0
1  41479.0       2.0
2  41481.0       2.0
3  41482.0       9.0
4  41483.0       2.0

### Health Care

HSQ_E
- HSD010 - General health condition
1 - Excellent
2 - Very good
3 - Good
4 - Fair
5 - Poor
7 - Refused, 9 - Don’t Know

HUQ_E
- HUQ010 - General health condition
1 - Excellent
2 - Very good
3 - Good
4 - Fair
5 - Poor
7 - Refused, 9 - Don’t Know
- HUQ041 / 040 - Type place most often go for healthcare
1- Clinic or health center
2 - Doctor's office or HMO
3 - Hospital emergency room
4 - Hospital outpatient department	
5 - Some other place
6 - Doesn't go to one place most often
77 - Refused, 99 - Don't know
- HUQ051/050 - #times receive healthcare over past year
0 - 8 are ranges, 77, 99
- HUQ090 - Seen mental health professional/past yr
1, 2, 7, 9


In [209]:
HUQ_2007_2008 = pd.read_csv('./Questionnaire/HUQ_E.csv')

HUQ_E = HUQ_2007_2008.filter(['SEQN', 'HUQ010', 'HUQ040','HUQ050'], axis=1)
HUQ_cols = {'SEQN':'SEQN', 'HUQ010': 'gen_health_cond','HUQ040':'place_healthcare',
            'HUQ050':'healthcare_past_yr'}

HUQ_E.columns = HUQ_E.columns.map(HUQ_cols)

In [210]:
HUQ_E.head()

SEQN  gen_health_cond  place_healthcare  healthcare_past_yr
0  41475.0              4.0               1.0        3.000000e+00
1  41476.0              1.0               2.0        1.000000e+00
2  41477.0              3.0               1.0        2.000000e+00
3  41478.0              3.0               2.0        2.000000e+00
4  41479.0              3.0               2.0        5.397605e-79

### Health Insurance

variables:
- HIQ011 : health_insurance - {Are you/Is SP} covered by health insurance or some other kind of health care plan? 1 yes, 2 no

In [211]:
HIQ_2007_2008 = pd.read_csv('./Questionnaire/HIQ_E.csv')

HIQ_E = HIQ_2007_2008.filter(['SEQN', 'HIQ011'], axis=1)
HIQ_cols = {'SEQN':'SEQN', 'HIQ011': 'health_insurance'}

HIQ_E.columns = HIQ_E.columns.map(HIQ_cols)

In [212]:
HIQ_E.head()

SEQN  health_insurance
0  41475.0               2.0
1  41476.0               1.0
2  41477.0               1.0
3  41478.0               1.0
4  41479.0               2.0

### Househole Characteristics

HOQ_E
- HOD050 - Number of rooms in home
1-13, 777, 999
- HOQ065 - Home owned, bought, rented, other
1 - Owned or being bought
2 - Rented
3 - Other arrangements
7 - Refused, 9 - Don't know


In [213]:
HOQ_2007_2008 = pd.read_csv('./Questionnaire/HOQ_E.csv')

HOQ_E = HOQ_2007_2008.filter(['SEQN', 'HOD050','HOQ065'], axis=1)
HOQ_cols = {'SEQN':'SEQN', 'HOD050': '#_rooms','HOQ065':'home_ownership'}

HOQ_E.columns = HOQ_E.columns.map(HOQ_cols)

In [214]:
HOQ_E.head()

SEQN  #_rooms  home_ownership
0  41475.0      4.0             1.0
1  41476.0      9.0             1.0
2  41477.0      6.0             1.0
3  41478.0      5.0             2.0
4  41479.0      5.0             1.0

### Early Childhood

ECQ_E
- ECQ020 - Mother smoked when pregnant
1=yes, 2=no, 7/9

In [215]:
ECQ_2007_2008 = pd.read_csv('./Questionnaire/ECQ_E.csv')

ECQ_E = ECQ_2007_2008.filter(['SEQN', 'ECQ020'], axis=1)
ECQ_cols = {'SEQN':'SEQN', 'ECQ020': 'mom_smoked_preg'}

ECQ_E.columns = ECQ_E.columns.map(ECQ_cols)

In [216]:
ECQ_E.head()

SEQN  mom_smoked_preg
0  41476.0              2.0
1  41478.0              1.0
2  41480.0              2.0
3  41484.0              2.0
4  41488.0              2.0

### Food Security

FSQ_E
- FSD032A - HH Worried run out of food
    - 1 - Often true
    - 2 - Sometimes true
    -  - Never true
    - 7 - Refused, 9 - Don't know
- FSD032B - HH Food didn't last
    - “ “
- FSD032C - HH Couldn't afford balanced meals
    - “ “
- FSD041 - HH Adults cut size or skip meals
    - 1, 2, 7, 9
- FSD032D - HH Relied on low-cost food for child
1 - Often true
2 - Sometimes true
3 - Never true
7 - Refused, 9 - Don't know
- FSDHH - Household food security category
1 - HH full food security: 0
2 - HH marginal food security: 1-2
3 - HH low food security: 3-5 (HH w/o child) / 3-7 (HH w/ child)
4 - HH very low food security: 6-10 (HH w/o child) / 8-18 (HH w/ child)
No 7/9, just missing
- FSQ012 / 171 - HH FS benefit: receive in last 12 months
1, 2, 7, 9
- FSQ162 - HH WIC benefit: receive in last 12 month
1,2,7,9


In [217]:
FSQ_2007_2008 = pd.read_csv('./Questionnaire/FSQ_E.csv')

FSQ_E = FSQ_2007_2008.filter(['SEQN', 'FSD032A', 'FSD032B', 'FSD032C', 'FSD041', 'FSD032D', 'FSDHH',
                             'FSQ171', 'FSQ162'], axis=1)

 
FSQ_cols = {'SEQN': 'SEQN', 'FSD032A':'run_out_food', 'FSD032B':'food_didnt_last', 'FSD032C':'afford_balanced_meals', 
             'FSD041':'adults_skip_meals', 'FSD032D':'relied_cheap_food', 'FSDHH':'food_security_cat',
             'FSQ171':'food_stamps', 'FSQ162':'WIC'}

FSQ_E.columns = FSQ_E.columns.map(FSQ_cols)

In [218]:
FSQ_E.head()

SEQN  run_out_food  food_didnt_last  afford_balanced_meals  \
0  41475.0           3.0              3.0                    3.0   
1  41476.0           3.0              3.0                    3.0   
2  41477.0           3.0              3.0                    3.0   
3  41478.0           2.0              2.0                    2.0   
4  41479.0           3.0              3.0                    3.0   

   adults_skip_meals  relied_cheap_food  food_security_cat  food_stamps  WIC  
0                NaN                NaN                1.0          NaN  NaN  
1                NaN                3.0                1.0          NaN  2.0  
2                NaN                NaN                1.0          NaN  NaN  
3                2.0                2.0                3.0          1.0  1.0  
4                NaN                3.0                1.0          NaN  1.0

### Income

INQ_E
- IND235 - Monthly family income
    - 1-12 ranges, 77, 99
- INQ244 - Family has savings more than $5000
    - 1,2,7,9


In [219]:
INQ_2007_2008 = pd.read_csv('./Questionnaire/INQ_E.csv')

INQ_E = INQ_2007_2008.filter(['SEQN', 'IND235', 'INQ244'], axis=1)

INQ_cols = {'SEQN': 'SEQN', 'IND235':'monthly_income', 'INQ244':'savings_5000'}

INQ_E.columns = INQ_E.columns.map(INQ_cols)


In [220]:
INQ_E.head()

SEQN  monthly_income  savings_5000
0  41475.0             5.0           2.0
1  41476.0            12.0           NaN
2  41477.0             4.0           2.0
3  41478.0             3.0           2.0
4  41479.0             7.0           NaN

### Medical Conditions

variables:
- MCQ010 : doc_asthma - Has a doctor or other health professional ever told {you/SP} that {you have/s/he/SP has} asthma (az-ma)?
- MCQ080 : doc_overweight - Has a doctor or other health professional ever told {you/SP} that {you were/s/he/SP was} overweight?
- MCQ160a :  doc_arthritis - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had arthritis (ar-thry-tis)?
- MCQ160b :  doc_CHF - Has a doctor or other health professional ever told {you/SP} that {you/s/he}had congestive heart failure?
- MCQ160c : doc_CHD - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had coronary (kor-o-nare-ee) heart disease?
- MCQ160e : doc_Eeart_attack - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had a heart attack (also called myocardial infarction (my-o-car-dee-al in-fark-shun))?
- MCQ160f : doc_stroke - Has a doctor or other health professional ever told {you/SP} that {you/s/her} had a stroke?
- MCQ160k : doc_chronic_bronchitis - Has a doctor or other health professional ever told {you/SP} that {you/s/he} . . .had chronic bronchitis?
- MCQ160l : doc_liver_condition - Has a doctor or other health professional ever told {you/SP} that {you/s/he} . . .had any kind of liver condition?
- MCQ160m : doc_thyroid_problem - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had another thyroid (thigh-roid) problem?
- MCQ160o : doc_COPD - Has a doctor or other health professional ever told {you/SP} that {you/s/he} had COPD?
- MCQ220 : doc_cancer - {Have you/Has SP} ever been told by a doctor or other health professional that {you/s/he} had cancer or a malignancy (ma-lig-nan-see) of any kind?
- MCQ365a : doc_lose_weight - To lower {your/SP's} risk for certain diseases, during the past 12 months {have you/has s/he} ever been told by a doctor or health professional to: control {your/his/her} weight or lose weight?
- MCQ365b : doc_exercise - To lower {your/SP's} risk for certain diseases, during the past 12 months {have you/has s/he} ever been told by a doctor or health professional to: increase {your/his/her} physical activity or exercise?
- MCQ053 - Taking treatment for anemia/past 3 mos
    - 1,2,7,9
- MCQ084 - Difficulties in thinking or remembering
    - 1, 2, 7, 9
- MCQ230a - What kind of cancer
    - 10-39 different kinds, 66 = more than 3, 77, 99


In [221]:
MCQ_2007_2008 = pd.read_csv('./Questionnaire/MCQ_E.csv')

MCQ_E = MCQ_2007_2008.filter(['SEQN', 'MCQ010', 'MCQ080', 'MCQ160A', 'MCQ160B', 'MCQ160C', 'MCQ160E',
 'MCQ160F', 'MCQ160K', 'MCQ160L', 'MCQ160M', 'MCQ160O', 'MCQ220','MCQ053',
                             'MCQ084','MCQ230A'], axis=1)

 
MCQ_cols = {'SEQN': 'SEQN', 'MCQ010':'doc_asthma', 'MCQ080':'doc_overweight', 'MCQ160A':'doc_arthritis', 
 'MCQ160B':'doc_CHF', 'MCQ160C':'doc_CHD', 'MCQ160E':'doc_heart_attack',
 'MCQ160F':'doc_stroke', 'MCQ160K':'doc_chronic_bronchitis', 'MCQ160L':'doc_liver_condition', 
 'MCQ160M':'doc_thyroid_problem', 'MCQ160O':'doc_COPD', 'MCQ220':'doc_cancer','MCQ053': 'anemia',
                             'MCQ084':'difficult_remembering','MCQ230A':'cancer_type'}

MCQ_E.columns = MCQ_E.columns.map(MCQ_cols)

In [222]:
MCQ_E.head()

SEQN  doc_asthma  doc_overweight  doc_arthritis  doc_CHF  doc_CHD  \
0  41475.0         1.0             1.0            1.0      1.0      2.0   
1  41476.0         2.0             NaN            NaN      NaN      NaN   
2  41477.0         2.0             2.0            1.0      2.0      2.0   
3  41478.0         2.0             NaN            NaN      NaN      NaN   
4  41479.0         2.0             2.0            2.0      2.0      2.0   

   doc_heart_attack  doc_stroke  doc_chronic_bronchitis  doc_liver_condition  \
0               1.0         2.0                     2.0                  2.0   
1               NaN         NaN                     NaN                  NaN   
2               2.0         2.0                     2.0                  2.0   
3               NaN         NaN                     NaN                  NaN   
4               2.0         2.0                     2.0                  2.0   

   doc_thyroid_problem  doc_cancer  anemia  cancer_type  
0                  1.0         2.0     2.0          NaN  
1                  NaN         NaN     2.0          NaN  
2                  2.0         2.0     2.0          NaN  
3                  NaN         NaN     2.0          NaN  
4                  2.0         2.0     2.0          NaN

### Kidney Conditions - Urology (KIQ_U_E)

variables
- KIQ022: doc_kidney - {Have you/Has SP} ever been told by a doctor or other health professional that {you/s/he} had weak or failing kidneys? Do not include kidney stones, bladder infections, or incontinence.

In [223]:
KIQ_U_2007_2008 = pd.read_csv('./Questionnaire/KIQ_U_E.csv')

KIQ_U_E = KIQ_U_2007_2008.filter(['SEQN', 'KIQ022'], axis=1)
KIQ_U_cols = {'SEQN':'SEQN', 'KIQ022': 'doc_kidney'}

KIQ_U_E.columns = KIQ_U_E.columns.map(KIQ_U_cols)

In [224]:
KIQ_U_E.head()

SEQN  doc_kidney
0  41475.0         2.0
1  41477.0         1.0
2  41479.0         2.0
3  41481.0         2.0
4  41482.0         2.0

### Occupation

OCQ_E
- OCD150 - Type of work done last week
1 - Working at a job or business
2 - With a job or business but not at work
3 - Looking for work, or
4 - Not working at a job or business?
7 - Refused, 9-  Don’t know
- OCQ180 - Hours worked last week at all jobs
    - Range 0-120 hours
    - 77777, 999999


In [225]:
OCQ_2007_2008 = pd.read_csv('./Questionnaire/OCQ_E.csv')

OCQ_E = OCQ_2007_2008.filter(['SEQN', 'OCD150','OCQ180'], axis=1)
OCQ_cols = {'SEQN':'SEQN', 'OCD150': 'work_type','OCQ180':'hrs_worked_week'}

OCQ_E.columns = OCQ_E.columns.map(OCQ_cols)

In [226]:
OCQ_E.head()

SEQN  work_type  hrs_worked_week
0  41475.0        4.0              NaN
1  41477.0        4.0              NaN
2  41479.0        1.0             50.0
3  41481.0        3.0              NaN
4  41482.0        1.0             40.0

### Oral Health

OHQ_E
- OHQ680 - Last yr embarrassed because of mouth
    - 1 very often to 5 never, 7, 9


In [227]:
OHQ_2007_2008 = pd.read_csv('./Questionnaire/OHQ_E.csv')

OHQ_E = OHQ_2007_2008.filter(['SEQN', 'OHQ680'], axis=1)
OHQ_cols = {'SEQN':'SEQN', 'OHQ680': 'embarassed_mouth'}

OHQ_E.columns = OHQ_E.columns.map(OHQ_cols)

In [228]:
OHQ_E.head()

SEQN  embarassed_mouth
0  41475.0               3.0
1  41477.0               5.0
2  41479.0               4.0
3  41481.0               5.0
4  41482.0               4.0

### Osteoperosis

variables:
- OSQ010A : broken_Eip - Has a doctor ever told {you/SP} that {you/s/he} had broken or fractured {your/his/her} . . .hip?
- OSQ060 : doc_osteoporosis - Has a doctor ever told {you/SP} that {you/s/he} had osteoporosis, sometimes called thin or brittle bones?

    - 1 yes, 2 no, 7 refused, 9 don't know

In [229]:
OSQ_2007_2008 = pd.read_csv('./Questionnaire/OSQ_E.csv')

OSQ_E = OSQ_2007_2008.filter(['SEQN', 'OSQ010A', 'OSQ060'], axis=1)
OSQ_cols = {'SEQN':'SEQN', 'OSQ010A': 'broken_hip', 'OSQ060':'doc_osteoporosis'}

OSQ_E.columns = OSQ_E.columns.map(OSQ_cols)

In [230]:
OSQ_2007_2008.columns

Index(['SEQN', 'OSQ010A', 'OSQ010B', 'OSQ010C', 'OSQ020A', 'OSQ020B',
       'OSQ020C', 'OSD030AA', 'OSQ040AA', 'OSD050AA',
       ...
       'OSQ140U', 'OSQ150', 'OSQ160A', 'OSQ160B', 'OSQ170', 'OSQ180', 'OSQ190',
       'OSQ200', 'OSQ210', 'OSQ220'],
      dtype='object', length=105)

In [231]:
OSQ_E.head()

SEQN  broken_hip  doc_osteoporosis
0  41475.0         2.0               2.0
1  41477.0         2.0               2.0
2  41479.0         2.0               2.0
3  41481.0         2.0               2.0
4  41482.0         2.0               2.0

### Physical Activity

variables:
- PAQ650 - Vigorous recreational activities - In a typical week {do you/does SP} do any vigorous-intensity sports, fitness, or recreational activities that cause large increases in breathing or heart rate like running or basketball for at least 10 minutes continuously? 1 yes 2 no
    - PAD200 - Vigorous activity over past 30 days ^^
- PAQ665 - Moderate recreational activities
    - PAD320 - Moderate activity over past 30 days ^^
- PAD680 : min_sedetary - The following question is about sitting at school, at home, getting to and from places, or with friends including time spent sitting at a desk, traveling in a car or bus, reading, playing cards, watching television, or using a computer. Do not include time spent sleeping. How much time {do you/does SP} usually spend sitting on a typical day?
- PAQ706 - Days physically active at least 60 min.
    - 1-7, 77 99
- PAQ710 - Hours watch TV or videos past 30 days
    - 1-5, 8=don’t watch, 77, 99
- PAQ715 - Hours use computer past 30 days (on average/day)
    - “ “
- PAQ722 - Any physical activity in past 7 days

In [232]:
PAQ_2007_2008 = pd.read_csv('./Questionnaire/PAQ_E.csv')

PAQ_E = PAQ_2007_2008.filter(['SEQN', 'PAQ650', 'PAD200', 'PAQ665', 'PAD320',
           'PAD680', 'PAQ706', 'PAQ710','PAQ715','PAQ722'], axis=1)

PAQ_cols = {'SEQN':'SEQN', 'PAQ650':'vigorous_activity', 'PAD200':'vigorous_activity', 'PAQ665':'moderate_activity',
           'PAD3205':'moderate_activity', 'PAD680':'min_sedetary', 'PAQ706':'days_60min_active', 
            'PAQ710':'hours_TV','PAQ715':'hours_computer','PAQ722':'activity_week'}

PAQ_E.columns = PAQ_E.columns.map(PAQ_cols)

In [233]:
PAQ_E.columns

Index(['SEQN', 'vigorous_activity', 'moderate_activity', 'min_sedetary'], dtype='object')

In [234]:
PAQ_E.head() # activity per week missing

SEQN  vigorous_activity  moderate_activity  min_sedetary
0  41475.0                2.0                2.0         720.0
1  41476.0                NaN                NaN           NaN
2  41477.0                2.0                2.0         240.0
3  41479.0                2.0                2.0         180.0
4  41480.0                NaN                NaN           NaN

### Physical Functioning

PFQ_E
- PFQ049 - Limitations keeping you from working
    - 1,2,7,9
- PFQ059 - Physical, mental, emotional limitations
    - 1,2,7,9
- PFQ061A - Managing money difficulty
    - 1 no difficulty to 4 unable to do, 5 don’t do it, 7, 9

Many other activities I could include

In [235]:
PFQ_2007_2008 = pd.read_csv('./Questionnaire/PFQ_E.csv')

PFQ_E = PFQ_2007_2008.filter(['SEQN', 'PFQ049', 'PFQ059', 'PFQ061A'], axis=1)

PFQ_cols = {'SEQN':'SEQN', 'PFQ049':'limit_work', 'PFQ059':'PME_limit', 'PFQ061A':'manage_$_difficult'}

PFQ_E.columns = PFQ_E.columns.map(PFQ_cols)

In [236]:
PFQ_E.head()

SEQN  limit_work  PME_limit  manage_$_difficult
0  41475.0         1.0        NaN                 1.0
1  41476.0         NaN        NaN                 NaN
2  41477.0         2.0        2.0                 1.0
3  41478.0         NaN        NaN                 NaN
4  41479.0         2.0        2.0                 NaN

### Sleeping Disorders

variables:
- SLQ060 : doc_sleeping_disorder - {Have you/Has SP} ever been told by a doctor or other health professional that {you have/s/he has} a sleep disorder?
- SLD010H - How much sleep do you get (hours)?
    - 2-12, 77, 99

In [237]:
SLQ_2007_2008 = pd.read_csv('./Questionnaire/SLQ_E.csv')

SLQ_E = SLQ_2007_2008.filter(['SEQN', 'SLQ060','SLD010H'], axis=1)
SLQ_cols = {'SEQN':'SEQN', 'SLQ060': 'doc_sleeping_disorder','SLD010H':'hrs_sleep'}

SLQ_E.columns = SLQ_E.columns.map(SLQ_cols)

In [238]:
SLQ_2007_2008.columns

Index(['SEQN', 'SLD010H', 'SLD020M', 'SLQ030', 'SLQ040', 'SLQ050', 'SLQ060',
       'SLQ070A', 'SLQ070B', 'SLQ070C', 'SLQ070D', 'SLQ080', 'SLQ090',
       'SLQ100', 'SLQ110', 'SLQ120', 'SLQ130', 'SLQ140', 'SLQ150', 'SLQ160',
       'SLQ170', 'SLQ180', 'SLQ190', 'SLQ200', 'SLQ210', 'SLQ220', 'SLQ230',
       'SLQ240'],
      dtype='object')

In [239]:
SLQ_E.head()

SEQN  doc_sleeping_disorder  hrs_sleep
0  41475.0                    2.0        6.0
1  41477.0                    2.0        8.0
2  41479.0                    2.0        6.0
3  41481.0                    2.0        6.0
4  41482.0                    2.0        8.0

### Smoking - Cigarette Use (SMQ_E)

variables:
- SMQ020 : smoker - These next questions are about cigarette smoking and other tobacco use. {Have you/Has SP} smoked at least 100 cigarettes in {your/his/her} entire life?

In [240]:
SMQ_2007_2008 = pd.read_csv('./Questionnaire/SMQ_E.csv')

SMQ_E = SMQ_2007_2008.filter(['SEQN', 'SMQ020'], axis=1)
SMQ_cols = {'SEQN':'SEQN', 'SMQ020': 'smoker'}

SMQ_E.columns = SMQ_E.columns.map(SMQ_cols)

In [241]:
SMQ_E.head()

SEQN  smoker
0  41475.0     2.0
1  41477.0     1.0
2  41479.0     2.0
3  41481.0     2.0
4  41482.0     1.0

In [242]:
SMQRTU_2007_2008 = pd.read_csv('./Questionnaire/SMQRTU_E.csv')

# 680 or 681
SMQRTU_E = SMQRTU_2007_2008.filter(['SEQN', 'SMQ680'], axis=1)
SMQ_cols = {'SEQN':'SEQN', 'SMQ680': 'smoked_5days'}

SMQRTU_E.columns = SMQRTU_E.columns.map(SMQ_cols)

In [243]:
SMQRTU_E.head()

SEQN  smoked_5days
0  41475.0           2.0
1  41477.0           2.0
2  41479.0           2.0
3  41481.0           2.0
4  41482.0           2.0

### Sexual Behavior (SXQ_E)

variables:
- SXQ292 : sexual_orientation_M - Do you think of yourself as...1 Heterosexual or straight (attracted to women), 2 Homosexual or gay (attracted to men), 3 Bisexual (attracted to men and women), 4 Something else, 5 Not sure
- SXQ294 : sexual_orientation_E - ""
- SXQ753 : doc_EPV - Has a doctor or other health care professional ever told you that you had human papillomavirus or HPV? 1 yes, 2 no, 7 refuse, 9 don't know
- SXD171 : lifetime_Eemale_partners - In your lifetime, with how many women have you had any kind of sex?
- SXD101 : lifetime_male_partners - In your lifetime, with how many men have you had any kind of sex?


In [244]:
SXQ_2007_2008 = pd.read_csv('./Questionnaire/SXQ_E.csv')

SXQ_E = SXQ_2007_2008.filter(['SEQN', 'SXQ292', 'SXQ294', 'SXD171', 'SXQ410','SXD101', 'SXQ753'], axis=1)
SXQ_cols = {'SEQN':'SEQN', 'SXQ292' : 'sexual_orientation_M', 'SXQ294' : 'sexual_orientation_F', 
           'SXD171' : 'lifetime_female_partners', 'SXD101' : 'lifetime_male_partners', 
            'SXQ410':'lifetime_male_partners',
            'SXQ753' : 'doc_HPV'}

SXQ_E.columns = SXQ_E.columns.map(SXQ_cols)


In [245]:
SXQ_2007_2008.columns

Index(['SEQN', 'SXAISC', 'SXQ021', 'SXD031', 'SXQ101', 'SXQ350', 'SXQ130',
       'SXD171', 'SXQ400', 'SXQ410', 'SXQ430', 'SXQ450', 'SXQ470', 'SXQ490',
       'SXQ510', 'SXQ530', 'SXQ550', 'SXQ570', 'SXQ590', 'SXQ600', 'SXQ610',
       'SXQ251', 'SXQ260', 'SXQ265', 'SXQ270', 'SXQ272', 'SXQ280', 'SXQ292',
       'SXQ294'],
      dtype='object')

In [246]:
SXQ_E.head()

SEQN  sexual_orientation_M  sexual_orientation_F  \
0  41475.0                   NaN                   NaN   
1  41479.0                   NaN                   NaN   
2  41481.0                   1.0                   NaN   
3  41482.0                   NaN                   NaN   
4  41483.0                   NaN                   NaN   

   lifetime_female_partners  lifetime_male_partners  
0                       NaN                     NaN  
1                       NaN                     NaN  
2                      60.0            5.397605e-79  
3                   77777.0            5.397605e-79  
4                       3.0            5.397605e-79

notes: may want to concat the # of partners and sexual orientation despite gender

### Taste & Smell

CSQ_E
- CSQ010 - Had problem with smell past 12 months?
    - 1,2,7,9
- CSQ080 - Had problem with taste past 12 months?
    - 1,2,7,9
- CSQ200 - Persistent cold/flu last 12 months?
    - “ “
- CSQ204 - Frequent nasal congestion in past 12 mth

In [247]:
# CSQ_2007_2008 = pd.read_csv('./Questionnaire/CSQ_E.csv')

# CSQ_E = CSQ_2007_2008.filter(['SEQN', 'CSQ010','CSQ080','CSQ200','CSQ204'], axis=1)
# CSQ_cols = {'SEQN':'SEQN', 'CSQ010': 'prob_smell','CSQ080':'prob_tatse',
#             'CSQ200':'persistent_cold','CSQ204':'nasal_congestion'}

# CSQ_E.columns = CSQ_E.columns.map(CSQ_cols)

In [248]:
# CSQ_E.head()

### Alcohol Use (ALQ_E)

Variables: 
- ALQ151 : alcoholic - Was there ever a time or times in {your/SP's} life when {you/he/she} drank {DISPLAY NUMBER} or more drinks of any kind of alcoholic beverage almost every day? 1 yes, 2 no

In [249]:
ALQ_2007_2008 = pd.read_csv('./Questionnaire/ALQ_E.csv')

ALQ_E = ALQ_2007_2008.filter(['SEQN', 'ALQ150'], axis=1)
ALQ_cols = {'SEQN':'SEQN', 'ALQ150': 'alcoholic'}

ALQ_E.columns = ALQ_E.columns.map(ALQ_cols)

In [250]:
ALQ_E.head()

SEQN  alcoholic
0  41475.0        2.0
1  41477.0        2.0
2  41479.0        2.0
3  41481.0        2.0
4  41482.0        1.0

### Weight History

WHQ_E
- WHQ030 - How do you consider your weight
    - 1 - Overweight
    - 2 - Underweight,
    - 3 - About the right weight?
    - 7 - Refused, 9 - Don’t Know
- WHQ070 - Tried to lose weight in past year
    - 1,2,7,9


In [251]:
WHQ_2007_2008 = pd.read_csv('./Questionnaire/WHQ_E.csv')

WHQ_E = WHQ_2007_2008.filter(['SEQN', 'WHQ030','WHQ070'], axis=1)
WHQ_cols = {'SEQN':'SEQN', 'WHQ030':'consider_weight','WHQ070':'tried_lose_weight'}

WHQ_E.columns = WHQ_E.columns.map(WHQ_cols)

In [252]:
WHQ_E.head()

SEQN  consider_weight  tried_lose_weight
0  41475.0              1.0                1.0
1  41477.0              1.0                2.0
2  41479.0              3.0                2.0
3  41481.0              3.0                2.0
4  41482.0              1.0                2.0

## Laboratory Data

### Cholesterol - HDL (HDL_E)

variables:
- LBDHDD : HDL - Direct HDL-Cholesterol (mg/dL), range 10-173

In [253]:
HDL_2007_2008 = pd.read_csv('./Laboratory/HDL_E.csv')

HDL_E = HDL_2007_2008.filter(['SEQN', 'LBDHDD'], axis=1)
HDL_cols = {'SEQN':'SEQN', 'LBDHDD': 'HDL'}

HDL_E.columns = HDL_E.columns.map(HDL_cols)

In [254]:
HDL_E.head()

SEQN   HDL
0  41475.0  47.0
1  41476.0  45.0
2  41477.0  34.0
3  41479.0  47.0
4  41480.0   NaN

### Cholesterol - LDL & Triglycerides (TRIGLY_E)

variables:

- LBDLDL - LBDLDL = ROUND(LBXTC-(LBXHDD+LBXTR/5) for LBXTR less than or equal to 400 mg/dL, and missing for LBXTR greater than 400 mg/dL, range 14-375
- LBXTR : triglyceride - Triglyceride (mg/dL), range 13-4233

In [255]:
TRIGLY_2007_2008 = pd.read_csv('./Laboratory/TRIGLY_E.csv')

TRIGLY_E = TRIGLY_2007_2008.filter(['SEQN', 'LBDLDL', 'LBXTR'], axis=1)
TRIGLY_cols = {'SEQN':'SEQN', 'LBDLDL':'LDL','LBXTR': 'triglyceride'}

TRIGLY_E.columns = TRIGLY_E.columns.map(TRIGLY_cols)

In [256]:
TRIGLY_2007_2008.columns

Index(['SEQN', 'WTSAF2YR', 'LBXTR', 'LBDTRSI', 'LBDLDL', 'LBDLDLSI'], dtype='object')

In [257]:
TRIGLY_E.head()

SEQN    LDL  triglyceride
0  41479.0  121.0          99.0
1  41481.0    NaN           NaN
2  41485.0  119.0         172.0
3  41486.0  110.0         233.0
4  41487.0  105.0         124.0

### Cholesterol - Total (TCHOL_E)

variables:
- LBXTC : tot_cholesterol - Total Cholesterol( mg/dL), range 69-813

In [258]:
TCHOL_2007_2008 = pd.read_csv('./Laboratory/TCHOL_E.csv')

TCHOL_E = TCHOL_2007_2008.filter(['SEQN', 'LBXTC'], axis=1)
TCHOL_cols = {'SEQN':'SEQN', 'LBXTC': 'tot_cholesterol'}

TCHOL_E.columns = TCHOL_E.columns.map(TCHOL_cols)

In [259]:
TCHOL_E.head()

SEQN  tot_cholesterol
0  41475.0            179.0
1  41476.0            143.0
2  41477.0            191.0
3  41479.0            188.0
4  41480.0              NaN

### Glycohemoglobin (GHB_E)

variables:
- LBXGH - Glycohemoglobin (%), range 3.5-17.5

In [260]:
GHB_2007_2008 = pd.read_csv('./Laboratory/GHB_E.csv')

GHB_E = GHB_2007_2008.filter(['SEQN', 'LBXGH'], axis=1)
GHB_cols = {'SEQN':'SEQN', 'LBXGH': 'glycohemoglobin'}

GHB_E.columns = GHB_E.columns.map(GHB_cols)

In [261]:
GHB_E.head()

SEQN  glycohemoglobin
0  41475.0              5.7
1  41477.0              9.5
2  41479.0              5.7
3  41481.0              NaN
4  41482.0              7.0

### Herpes Simplex Virus Type-1 & Type-2 (HSV_E)

variables:
- LBXHE2 : herpes_2 - Herpes Simplex Virus Type 2

In [262]:
HSV_2007_2008 = pd.read_csv('./Laboratory/HSV_E.csv')

HSV_E = HSV_2007_2008.filter(['SEQN', 'LBXHE2'], axis=1)
HSV_cols = {'SEQN':'SEQN', 'LBXHE2': 'herpes_2'}

HSV_E.columns = HSV_E.columns.map(HSV_cols)

In [263]:
HSV_E.head(2)

SEQN  herpes_2
0  41481.0       NaN
1  41485.0       2.0

### HIV Antibody Test (HIV_E)

variables:
- LBDHI : HIV - HIV antibody test result, 1 pos, 2, neg, 3 indeterminant

In [264]:
HIV_2007_2008 = pd.read_csv('./Laboratory/HIV_E.csv')

HIV_E = HIV_2007_2008.filter(['SEQN', 'LBDHI'], axis=1)
HIV_cols = {'SEQN':'SEQN', 'LBDHI': 'HIV'}

HIV_E.columns = HIV_E.columns.map(HIV_cols)

In [265]:
HIV_E.head(2)

SEQN  HIV
0  41481.0  NaN
1  41485.0  2.0

### High-Sensitivity C-Reactive Protein (hs-CRP) (HSCRP_I) - NOT IN 2013-2014

variables:
- LBXHSCRP : CRP - HS C-Reactive Protein (mg/L), values 0-188, missing values are mostly below detection limit --> can be made to 0

In [266]:
# HSCRP_2007_2008 = pd.read_csv('./Laboratory/HSCRP_E.csv')

# HSCRP_I = HSCRP_2015_2016.filter(['SEQN', 'LBXHSCRP '], axis=1)
# HSCRP_cols = {'SEQN':'SEQN', 'LBXHSCRP ': 'CRP'}

# HSCRP_I.columns = HSCRP_I.columns.map(HSCRP_cols)

### Serums

CUSEZN_E
- LBXSSE - Serum Selenium (ug/L) - cont range, missing
- LBXSZN - Serum Zinc (ug/dL) - continuous range, missing
- URXUCR - Urinary creatinine (mg/dL) - continuous range, missing

In [267]:
# CUSEZN_2007_2008 = pd.read_csv('./Laboratory/CUSEZN_E.csv')

# CUSEZN_E = CUSEZN_2007_2008.filter(['SEQN', 'LBXSSE','LBXSZN','URXUCR'], axis=1)
# CUSEZN_cols = {'SEQN':'SEQN', 'LBXSSE': 'serum_selenium', 'LBXSZN':'serum_zinc','URXUCR':'serum_creatinine'}

# CUSEZN_E.columns = CUSEZN_E.columns.map(CUSEZN_cols)

In [268]:
# CUSEZN_E.head()

### Folate

FOLATE_E
- LBDRFO - RBC folate (ng/mL) -  continuous range, missing

FOLFMS_E
- LBDFOT - Serum total folate (ng/mL) -  continuous range, missing

In [269]:
FOLATE_2007_2008 = pd.read_csv('./Laboratory/FOLATE_E.csv')
#FOLFMS_2007_2008 = pd.read_csv('./Laboratory/FOLFMS_E.csv')

# LBDRBF or LBDRFO
FOLATE_E = FOLATE_2007_2008.filter(['SEQN', 'LBDRBF', 'LBDFOL'], axis=1)
FOLATE_cols = {'SEQN':'SEQN', 'LBDRBF':'rbc_folate','LBDFOL':'serum_folate'}
#FOLFMS_E = FOLFMS_2007_2008.filter(['SEQN', 'LBDFOT'], axis=1)
#FOLFMS_cols = {'SEQN':'SEQN', 'LBDFOT':'serum_Eolate'}

FOLATE_E.columns = FOLATE_E.columns.map(FOLATE_cols)
#FOLFMS_E.columns = FOLFMS_E.columns.map(FOLFMS_cols)

In [270]:
# FOLFMS_E.head()

In [271]:
FOLATE_E.head()

SEQN  rbc_folate  serum_folate
0  41475.0      1099.3          24.0
1  41476.0       396.0          14.3
2  41477.0       613.7          29.0
3  41478.0       587.2          29.1
4  41479.0       582.8          36.9

### Fasting Glucose

GLU_E
- LBXGLU - Fasting Glucose (mg/dL) - 51 to 421, missing

In [272]:
GLU_2007_2008 = pd.read_csv('./Laboratory/GLU_E.csv')

GLU_E = GLU_2007_2008.filter(['SEQN', 'LBXGLU'], axis=1)
GLU_cols = {'SEQN':'SEQN', 'LBXGLU': 'fasting_glucose'}

GLU_E.columns = GLU_E.columns.map(GLU_cols)

In [273]:
GLU_E.head()

SEQN  fasting_glucose
0  41479.0            113.0
1  41481.0              NaN
2  41485.0             97.0
3  41486.0            123.0
4  41487.0            108.0

### Vitamins B12 and D

VITB12_E
- LBDB12 - Vitamin B12(pg/mL) - range, missing

VID_E
- LBXVIDMS - 25OHD2+25OHD3 (nmol/L) - range, missing (vitamin D)

In [274]:
# VITB12_2007_2008 = pd.read_csv('./Laboratory/VITB12_E.csv')
#  # LBXB12 or LBDB12
# VITB12_E = VITB12_2007_2008.filter(['SEQN', 'LBXB12'], axis=1)
# VITB12_cols = {'SEQN':'SEQN', 'LBXB12': 'lab_VitB12'}

# VITB12_E.columns = VITB12_E.columns.map(VITB12_cols)


VID_2007_2008 = pd.read_csv('./Laboratory/VID_E.csv')

VID_E = VID_2007_2008.filter(['SEQN', 'LBXVIDMS'], axis=1)
VID_cols = {'SEQN':'SEQN', 'LBXVIDMS': 'lab_VitD'}

VID_E.columns = VID_E.columns.map(VID_cols)


In [275]:
# VITB12_E.head()

In [276]:
VID_E.head()

SEQN  lab_VitD
0  41475.0      58.8
1  41476.0      80.9
2  41477.0      81.8
3  41478.0       NaN
4  41479.0      78.4

## Examination Data

### Blood Pressure (BPX_E)

variables:
- BPXSY1 : systolic_BP - Systolic: Blood pres (1st rdg) mm Hg
- BPXDI1 : diastolic_BP - Diastolic: Blood pres (1st rdg) mm Hg
- BPXPLS - 60 sec. pulse (30 sec. pulse * 2) - 0 to 180, missing

In [277]:
BPX_2007_2008 = pd.read_csv('./Examination/BPX_E.csv')

BPX_E = BPX_2007_2008.filter(['SEQN', 'BPXSY1', 'BPXDI1','BPXPLS'], axis=1)
BPX_cols = {'SEQN':'SEQN', 'BPXSY1': 'systolic_BP', 'BPXDI1' : 'diastolic_BP','BPXPLS':'pulse'}

BPX_E.columns = BPX_E.columns.map(BPX_cols)

In [278]:
BPX_E.head(2)

SEQN  systolic_BP  diastolic_BP  pulse
0  41475.0        128.0          64.0   66.0
1  41476.0          NaN           NaN    NaN

### Body Measures (BMX_E)

variables:
- BMXBMI : BMI - Body Mass Index (kg/m**2), range 12.1 to 82.9
- BMXWAIST : waist_C - Waist Circumference (cm), 40.2 to 177.9

In [279]:
BMX_2007_2008 = pd.read_csv('./Examination/BMX_E.csv')

BMX_E = BMX_2007_2008.filter(['SEQN', 'BMXBMI', 'BMXWAIST'], axis=1)
BMX_cols = {'SEQN':'SEQN', 'BMXBMI': 'BMI', 'BMXWAIST' : 'waist_C'}

BMX_E.columns = BMX_E.columns.map(BMX_cols)

In [280]:
BMX_E.head(2)

SEQN    BMI  waist_C
0  41475.0  58.04    156.3
1  41476.0  15.18     52.7

### Dual-Energy X-ray Absorptiometry - Whole Body (DXX_E)

variables:
- DXDTOBMD : bone_mineral_density - Total Bone Mineral Density (g/cm^2), range 0.559 to 1.548

In [281]:
# DXX_2007_2008 = pd.read_csv('./Examination/DXX_E.csv')

# DXX_E = DXX_2007_2008.filter(['SEQN', 'DXDTOBMD'], axis=1)
# DXX_cols = {'SEQN':'SEQN', 'DXDTOBMD': 'bone_mineral_density'}

# DXX_E.columns = DXX_E.columns.map(DXX_cols)

In [282]:
# DXX_E.head(2)

### Oral Health

OHXREF_E
- OHAREC - Overall recommendation for care
    - 1 - See a dentist immediately
    - 2 - See a dentist within the next 2 weeks	
    - 3 - See a dentist at your earliest convenience
    - 4 - Continue your regular routine care
    - Missing


In [283]:
# OHXREF_2007_2008 = pd.read_csv('./Examination/OHXREF_E.csv')

# OHXREF_E = OHXREF_2007_2008.filter(['SEQN', 'OHAREC'], axis=1)
# OHXREF_cols = {'SEQN':'SEQN', 'OHAREC': 'oral_rec'}

# OHXREF_E.columns = OHXREF_E.columns.map(OHXREF_cols)

In [284]:
# OHXREF_E.head()

# Merging dataframes

In [285]:
from functools import reduce

# list of df
# minus DLQ_E, CSQ_E, DXX_E, CUSEZN_E, FOLFMS_E, VITB12_E, OHXREF_E
data_Erames = [DPQ_E, OSQ_E, DEMO_E, DIET_E, DIQ_E, DBQ_E, DUQ_E, HUQ_E, HIQ_E, 
               HOQ_E, ECQ_E, FSQ_E, INQ_E, MCQ_E, KIQ_U_E, OCQ_E, OHQ_E, 
               PAQ_E, PFQ_E, SLQ_E, SMQ_E, SMQRTU_E, SXQ_E, ALQ_E, WHQ_E, 
               HDL_E, TRIGLY_E, TCHOL_E, GHB_E, HSV_E, HIV_E,
               FOLATE_E, GLU_E, VID_E, BPX_E, BMX_E]

df_2007_2008 = reduce(lambda  left,right: pd.merge(left,right,on=['SEQN'],
                                            how='outer'), data_Erames)

In [286]:
df_2007_2008.depressed

0        0.0
1        0.0
2        0.0
3        0.0
4        1.0
        ... 
10144    NaN
10145    NaN
10146    NaN
10147    NaN
10148    NaN
Name: depressed, Length: 10149, dtype: float64

In [287]:
df_2007_2008.dropna(subset=['depressed'], inplace=True)

In [288]:
df_2007_2008.shape

(5447, 100)

In [289]:
df_2007_2008.columns

Index(['SEQN', 'depressed', 'broken_hip', 'doc_osteoporosis', 'race_ethnicity',
       'edu_level', '#_ppl_household', 'age', 'gender', 'annual_HI',
       'marital_status', 'caffeine', 'special_diet', 'protein', 'carb',
       'sugar', 'fiber', 'fat', 'diet_chol', 'VitB6', 'folate', 'VitB12',
       'VitD', 'magnesium', 'iron', 'zinc', 'selenium', 'water',
       'doc_diabetes', 'doc_prediabetes', '#meals_fast_food',
       'how_healthy_diet', 'used_CMH', 'gen_health_cond', 'place_healthcare',
       'healthcare_past_yr', 'health_insurance', '#_rooms', 'home_ownership',
       'mom_smoked_preg', 'run_out_food', 'food_didnt_last',
       'afford_balanced_meals', 'adults_skip_meals', 'relied_cheap_food',
       'food_security_cat', 'food_stamps', 'WIC', 'monthly_income',
       'savings_5000', 'doc_asthma', 'doc_overweight', 'doc_arthritis',
       'doc_CHF', 'doc_CHD', 'doc_heart_attack', 'doc_stroke',
       'doc_chronic_bronchitis', 'doc_liver_condition', 'doc_thyroid_problem',
     

# Export 2007-2008 data

In [290]:
df_2007_2008.to_csv('/Users/carolinesklaver/Desktop/Capstone/NHANES/data/csv_data/df_2007_2008_ext.csv', index=False)